### PR TITLE
Add e2e test for delegate-module and federated-npm

### PR DESCRIPTION
**Pull request by VladyslavZinchenko**
image
image

### CODE REVIEW

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+appsData.forEach(
+    function(
+        property: { 
+            headerSelector: string
+            subHeaderSelector: string
+            buttonSelector: string,
+            headerText: string,
+            appNameText: string,
+            buttonNameText: string,
+            host: number
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: You can left there only `appNameText` and `host`. Rest of attr you can use directly in your testcase

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }) {
+    let host = property.host === 3002 ? appsData[1].host : appsData[0].host;
+    let appName = property.host === 3002 ? appsData[1].appNameText : appsData[0].appNameText;
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: It can be removed

File: federated-npm/e2e/checkApplications.cy.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+appsData.forEach(
+    function(
+        property: { 
+            headerSelector: string
+            subHeaderSelector: string
+            buttonSelector: string,
+            headerText: string,
+            appNameText: string,
+            buttonNameText: string,
+            host: number
```
Comment (PeaceDate) [inline federated-npm/e2e/checkApplications.cy.ts]: The same

File: federated-npm/e2e/checkApplications.cy.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }) {
+        const appName = property.host === 3001 ? appsData[0].appNameText : property.host === 3002 ? appsData[1].appNameText : appsData[2].appNameText;
+        const host = property.host === 3001 ? appsData[0].host : property.host === 3002 ? appsData[1].host : appsData[2].host;
```
Comment (PeaceDate) [inline federated-npm/e2e/checkApplications.cy.ts]: The same

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+
+    describe(`Check ${appName}`, () => {
+  
+        it(`Check ${appName} built and running`, () => {
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: Make name of test more detailed

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                selector: property.headerSelector,
+                text: property.headerText
+            })
+            basePage.checkElementWithTextPresence({
+                selector: property.subHeaderSelector,
+                text: `${appName}`
+            })
+        })
+    
+        it(`Check buttons in ${appName} exist`, () => {
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: The same

File: federated-npm/e2e/checkApplications.cy.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+
+    describe(`Check ${appName}`, () => {
+  
+        it(`Check ${appName} built and running`, () => {
```
Comment (PeaceDate) [inline federated-npm/e2e/checkApplications.cy.ts]: The same

File: federated-npm/e2e/checkApplications.cy.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                selector: property.headerSelector,
+                text: property.headerText
+            })
+            basePage.checkElementWithTextPresence({
+                selector: property.subHeaderSelector,
+                text: `${appName}`
+            })
+        })
+    
+        it(`Check buttons in ${appName} exist`, () => {
```
Comment (PeaceDate) [inline federated-npm/e2e/checkApplications.cy.ts]: The same

**Review by PeaceDate — CHANGES_REQUESTED** [review]

File: federated-npm/e2e/checkApplications.cy.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            basePage.openLocalhost(host)
+            basePage.checkElementWithTextPresence({
```
Comment (edchai) [inline federated-npm/e2e/checkApplications.cy.ts:47]: always check visibility if you can, visibilityState: 'be.visible'

**Review by edchai — CHANGES_REQUESTED** [review]

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+]
+
+appsData.forEach(
+    function(
+        property: { 
+            appNameText: string,
+            host: number
+    }) {
+    let host = property.host === 3002 ? appsData[1].host : appsData[0].host;
+    let appName = property.host === 3002 ? appsData[1].appNameText : appsData[0].appNameText;
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: Forgot to remove

File: delegate-modules/e2e/checkApplications.cy.ts [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import { BaseMethods } from '../../cypress/common/base';
+import { baseSelectors } from '../../cypress/common/selectors';
+import { Constants } from '../../cypress/fixtures/constants';
+
+const basePage: BaseMethods = new BaseMethods()
+
+let appsData = [
```
Comment (PeaceDate) [inline delegate-modules/e2e/checkApplications.cy.ts]: Use const instead of let

File: federated-npm/e2e/checkApplications.cy.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
+import { BaseMethods } from '../../cypress/common/base';
+import { baseSelectors } from '../../cypress/common/selectors';
+import { Constants } from '../../cypress/fixtures/constants';
+
+const basePage: BaseMethods = new BaseMethods()
+
+let appsData = [
```
Comment (PeaceDate) [inline federated-npm/e2e/checkApplications.cy.ts]: Use const instead of let

**Review by PeaceDate — CHANGES_REQUESTED** [review]

**Review by PeaceDate — APPROVED** [review]

**Review by edchai — APPROVED** [review]

**Pull request merged**